contents/computerVision/cv.md

```diff
@@ -0,0 +1,314 @@
+# bottom-right, bottom-left, to represent the portion of the
+# screen in the original image.
+(tl, tr, br, bl) = rect
+
+# Arbitrarily choose dimensions for the output image.
+width = 100
+height = 100
+
+# Construct an array to store the desired coordinates of the 4
+# corners of the screen after being transformed.
+dst = numpy.array([
+	[0, 0],
+	[width - 1, 0],
+	[width - 1, height - 1],
+	[0, height - 1]], dtype = "float32")
+
+# Calculate the perspective transform matrix and warp the
+# portion of the original image to grab the up-right screen.
+M = cv2.getPerspectiveTransform(rect, dst)
+warp = cv2.warpPerspective(orig, M, (width, height))
+# 'orig' here refers to the original image.
+# 'warp' now stores the screen projected straight and up-right
+# as seen in the picture above.
+```
+
+### Image Classification
+
+Image classification refers to the task of identifying the object present in an image. Typically, a predetermined set of possible objects is given and the classification of the image is the object that is most likely to exist in it.<sup>[[6]](#footnote6)</sup>
+
+![Classification](image-classification-1.png "Cat Classified")
+
+*[source](http://cs231n.github.io/classification/)*
+
+Image classification is often paired with object localization, which is the task of finding where in the image the object is. Many more complex problems in computer vision, such as object detection and segmentation, can be reduced to image classification and localization.<sup>[[6]](#footnote6)</sup>
+
+![Classification](image-classification-2.png "Cat Localized")
+
+*[Fei-Fei Li, Andrej Karpathy & Justin Johnson (2016) cs231n, Lecture 8 - Slide 8, Spatial Localization and Detection (01/02/2016)](http://cs231n.github.io/classification/)*
+
+At the moment, there already exist classification/localization models that surpass trained humans. For example, in the 2014 ImageNet Large Scale Visual Recognition Challenge (ILSVRC),
+> Our results indicate that a trained human annotator is capable of outperforming the best model (GoogLeNet) with approximately 1.7% chance (p = 0.022).<sup>[[8]](#footnote8)</sup>
+
+**Example:** Classifying images using deep learning and OpenCV
+
+*This example is extracted from "Deep Learning with OpenCV" tutorial originally published on PyImageSearch on August 21, 2017. Refer to original tutorial [here](https://www.pyimagesearch.com/2017/08/21/deep-learning-with-opencv/). All pictures and code snippets in this example belong to its original author, Adrian Rosebrock.*
+
+In this example, we use OpenCV and a pre-trained image classification model called the GoogLeNet. Refer to GoogLeNet's original paper [here](https://www.cs.unc.edu/~wliu/papers/GoogLeNet.pdf). The goal of this example is to use the model to classify the object present in a given image. First, we load the image in question and the model, then we feed the image to the model and take the most confident prediction.
```

Comment:
May be put the paper info in references (and cite from main text) rather than in the main text itself?